### PR TITLE
[RFC] Adds Ping() to client/scheduler/server

### PR DESCRIPTION
**Pull request by pbarnum**

### CODE REVIEW

**Comment by pbarnum** [comment]
@hibiken, this should close #445

**Comment by mdaliyan** [comment]
Right now I'm pushing a task with an empty payload and then I'm ignoring it when I consume it for the sake of health check. Exposing Ping makes it much easier and cleaner health check than my approach. Please consider increasing the priority for this PR if possible. Thank you.

**Review by rapita — APPROVED** [review]

**Comment by bojanz** [comment]
@hibiken What can we do to push this along? I need a client.Ping() to be able to integrate into my app's health check metrics.

**Comment by arpitflex** [comment]
We are deploying an asynq queue alongside our services in a Kubernetes cluster, and a readiness check like this would be very helpful to ensure better Pod restarts by the orchestrator. Are we looking to include this with the 0.25 release @hibiken?

File: scheduler.go [hunk 1]
```diff
@@ -320,3 +320,8 @@ func (s *Scheduler) clearHistory() {
 		}
 	}
 }
+
+// Ping performs a ping against the redis connection.
+func (s *Scheduler) Ping() error {
```
Comment (kamikazechaser) [inline scheduler.go:325]: Check for s.srvStateClosed here as well

**Review by kamikazechaser — CHANGES_REQUESTED** [review]

**Pull request merged**